articles/password-checker-javascript/index.md

```diff
@@ -0,0 +1,250 @@
+
+- The third element is a `<span>` which is where the password strength will be labeled as a badge with color: Green for strong, Blue for medium and Red for weak.
+
+### JavaScript Code
+
+Let's create five variables: 
+
+- `timeout`: for referencing the timeout before a callback is called.
+
+- `password`: for the input where the password will be entered.
+
+- `strengthBadge`: for storing the span used for displaying the strength of the password (which we can access using the DOM element's ID.
+
+- `strongPassword` & `mediumPassword`: for storing the Regex conditions.
+
+```JavaScript
+let timeout;
+let password = document.getElementById('PassEntry');
+let strengthBadge = document.getElementById('StrengthDisp');
+let strongPassword = new RegExp('(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{8,})')
+let mediumPassword = new RegExp('((?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{6,}))|((?=.*[a-z])(?=.*[A-Z])(?=.*[^A-Za-z0-9])(?=.{8,}))')
+```
+
+Let's create a function called `StrengthChecker` where we test for the strength of the passwords using the regular expressions that we wrote with the `RegExp.prototype.test()` method. It returns `true` if there is a match or `false` if there is no match.
+
+Then, let's set the background colour (`strengthBadge.style.backgroundColor`) and text of the badge (`strengthBadge.textContent`) according to the result. 
+
+```JavaScript
+function StrengthChecker(PasswordParameter) {
+    if(strongPassword.test(PasswordParameter)) {
+        strengthBadge.style.backgroundColor = "green";
+        strengthBadge.textContent = 'Strong';
+    } else if(mediumPassword.test(PasswordParameter)) {
+        strengthBadge.style.backgroundColor = 'blue';
+        strengthBadge.textContent = 'Medium';
+    } else {
+        strengthBadge.style.backgroundColor = 'red';
+        strengthBadge.textContent = 'Weak';
+    }
+}
+```
+
+Let's add an input event listener to the `password` input and check the strength with the `StrengthChecker` function when the user has typed something. We won't call the function immediately after every keystroke. When the user types quickly, we should wait until a pause occurs. So, instead of checking the strength immediately, we'll set a timeout.
+
+We should clear the previous timeout if there is any. When the events occur close together than our timeout duration, the timeout from the preceding input event should be canceled.
+
+```JavaScript
+password.addEventListener("input", () => {
+    strengthBadge.style.display = 'block';
+    clearTimeout(timeout);
+    timeout = setTimeout(() => StrengthChecker(password.value), 500);
+    if(password.value.length !== 0) {
+        strengthBadge.style.display != 'block';
+    } else {
+        strengthBadge.style.display = 'none';
+    }
+});
+```
+
+The full JavaScript code:
+
+```JavaScript
+<script>
+    
```

Comment:
I'd suggest you create a new file for the JS code instead of writing it in a Script tag to keep things clean. 

You should instruct the users to create a new file to write the code and then link it using a script tag.